### PR TITLE
 Bug 1860814 -  fix amo_prod__desktop_addons_by_client

### PR DESCRIPTION
**Pull request by lelilia**
[Bug 1860814](https://bugzilla.mozilla.org/show_bug.cgi?id=1860814)

The problem is that we have one very active customer_id with lot's of entries in main 
with the previous code all the copies for the addons that come with every row needed to be aggregated.
By adding a step in the ETL where I only get the distinct addons entries for a customer_id the aggregation in the next step is not so big that it exceeds the query limits any longer 

This is not the most elegant solution, so I'm very open to feedback on how to achieve the same result in a better way.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-1864)

### CODE REVIEW

**Review by Iinh — COMMENTED** [review]
Considering that the DAG failed due to resource constraints, rewriting the query this way to fix the failure makes sense to me. However, because this is a tier-1 DAG and I'm not familiar with it, I'd appreciate it if another engineer with more context could also review these changes before we ship.

File: sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql [hunk 1]
```diff
@@ -57,7 +43,41 @@ WITH per_client AS (
   GROUP BY
     submission_date,
     sample_id,
-    client_id
+    client_id,
+    addons
+),
+per_client AS (
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql:49]: Unfortunately this is not per_client, it is per_client/app-version/country/locale/app_os. If a user has pings with different values for e.g. country _and_ different `payload.info.addons`, we'll end up with multiple rows for that client in this CTE.

Is there any way we can instead pull this information from `clients_daily_v6`? That table has addon information, and probably already has this other data aggregated.

**Review by fbertsch — CHANGES_REQUESTED** [review]

File: sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql (hunk 1 above)
Comment (lelilia) [reply sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql:49]: oh you are right ... how about moving that step out and joining it later? 
The problem with `clients_daily_v6` would be that they get the addons data from environment.addons while this query gets it from `payload.info.addons` (which is deprecated as of now) but they values are not the same and so switching would change the outcome ... which we might want to do at some point but maybe not for a triage fix

File: sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql [hunk 2]
```diff
@@ -58,6 +43,65 @@ WITH per_client AS (
...
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    client_id,
+    sample_id,
+    payload.info.addons
+  FROM
+    telemetry.main
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND client_id IS NOT NULL
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql:57]: One nit: I'd move this `FROM` and `WHERE` clause to a new CTE that is queried by both `per_clients_just_addons_base` and `per_clients_without_addons`. That way we only define the filter/source once.

**Review by fbertsch — APPROVED** [review]
This LGTM, thank you!

**Pull request merged**